<?xml version="1.0"?>

<launch>
  <!-- This launch file is intended to aid in recording a path at Richmond Field Station.  State publishing and GPS plotting are enabled. -->

  <param name="vehicle/scripts_dir" type="str" value="$(find mkz_mpc_path_follower)/scripts/" />
  <!-- The trajectory can be generated using recorded time or a target velocity. -->
  <!-- Set track_using_time to True to allow for time-varying velocity. -->
  <!-- Set track_using_time to False to use target_vel (m/s) instead. -->
  <param name="vehicle/track_using_time" type="bool" value="True" />
  <param name="vehicle/target_vel" type="double" value="1.0" />

  <!-- Waypoints CSV to specify path. -->
  <!-- Set is_heading_info to true if Heading (N = 0) is used instead of Yaw (E = 0). -->
  <param name="vehicle/mat_waypoints" type="str" value="$(find mkz_mpc_path_follower)/paths/path1_6_20.mat" />
  <param name="vehicle/is_heading_info" type="bool" value="False"/>
  <param name="vehicle/time_check_on" type="bool" value="False"/>

  <!-- RFS Coordinate System Definition -->
  <param name="vehicle/lat0" type="double" value="37.917929"/>
  <param name="vehicle/lon0" type="double" value="-122.331798"/>
  <param name="vehicle/yaw0" type="double" value="0.0"/>

  <!--...................................................................-->

  <!-- State Publishing and Plotting Nodes -->
  <node ns="vehicle" pkg="mkz_mpc_path_follower" type="state_publisher.py" name="state_pub_node" output="log"/>
  <node ns="vehicle" pkg="mkz_mpc_path_follower" type="gps_plotter.py" name="plot_node" output="log"/>

  <!-- Recording Node:: TODO FIX TOPICS -->
  <arg name="can_topics" default=""/>
  <arg name="gps_topics" default=""/>
  <arg name="vehicle_topics" default="/vehicle/state_est /vehicle/target_path /vehicle/mpc_path"/>
  <arg name="perception_topics" default=""/>

  <node pkg="rosbag" type="record" name="rosbag_path_recorder" respawn="true" output="screen" 
        args="-o $(find mkz_mpc_path_follower)/bags/path_record $(arg can_topics) $(arg gps_topics) $(arg vehicle_topics) $(arg perception_topics)" />
</launch>
